streamer/src/recvmmsg.rs

```diff
@@ -107,17 +107,35 @@ pub fn recv_mmsg(sock: &UdpSocket, packets: &mut [Packet]) -> io::Result</*num p
             iov_len: buffer.len(),
         });
 
-        hdr.write(mmsghdr {
-            msg_len: 0,
-            msg_hdr: msghdr {
+        let mut msg_hdr: msghdr;
```

Comment:
This triggers a warning in non-musl builds because it doesn't need to be mutable. Let's just add an `#[allow(unused_mut)]` on top with a comment.

Our other options are to replicate all of the code under each block (have `hdr.write(...)` under the `cfg`), or to have a function to return the `msghdr`, but I think those will be easier to break in the future. What do you think?

```suggestion
        // Mutable for musl builds, see https://github.com/rust-lang/libc/issues/2344 for more info
        #[allow(unused_mut)]
        let mut msg_hdr: msghdr;
```

Reply:
Yeah, I think the allow directive makes the most sense here. It limits the amount of duplication.

Having a function could also work, not sure it would be any more fragile. Will give it a shot and push an update.

Reply:
Just pushed a change which I think fixes the lint warning without necessarily making things too complicated.